### PR TITLE
External Media: Add API unit tests

### PR DESCRIPTION
**Pull request by obenland**
Adds unit tests for external media API endpoints.

#### Changes proposed in this Pull Request:

* Adds unit tests for external media API endpoints

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Part of a new feature that allows users access to a free image library and their Google Photos account from within blocks.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:

To run unit tests:
* Follow [setup guidelines](https://github.com/Automattic/jetpack/blob/master/docs/development-environment.md#unit-testing) if you don't have unit tests set up yet.
* The `/copy` endpoint test is being skipped at the moment.
* Run `phpunit --filter WP_Test_WPCOM_REST_API_V2_Endpoint_External_Media`.

#### Proposed changelog entry for your changes:

Not necessary

### CODE REVIEW

**Comment by obenland** [comment]
@jeherve Attempting to make #15717 somewhat easier to review, I split out the API endpoint and added unit test (to make it easier to test without a client).

I've not had much luck creating a test for the image sideload endpoint. Not sure how to best mock an external image. Do you happen to have any advice or pointers there? Or is it fine to skip for now?

How can I tell Codeclimate about the added test coverage?

**Comment by obenland** [comment]
Closing in favor of merging #15717.

**Comment by obenland** [comment]
@jeherve How do you feel about introducing skipped tests?

**Comment by jeherve** [comment]
> How do you feel about introducing skipped tests?

I think that'd be fine for now, but maybe add a comment in there explaining the things you've tried and how that turned out, or maybe a link to a comment here with more details? This may help the next person who looks at this, and save them some time in their investigation?

**Comment by obenland** [comment]
I was able to make the image copy test work with bunch of filter modifications. Feels fragile, but maybe that's just how it needs to be when we're trying to test an endpoint for a remote image with a local file. 🤷‍♂️

All tests are 🟢  now, though!

File: tests/php/core-api/wpcom-endpoints/test_class-wpcom-rest-api-v2-endpoint-external-media.php.php [hunk 1]
```diff
@@ -0,0 +1,249 @@
+<?php // phpcs:ignore
```
Comment (frontdevde) [inline tests/php/core-api/wpcom-endpoints/test_class-wpcom-rest-api-v2-endpoint-external-media.php.php:1]: Minor: I assume it's unintentional that this file ends in `.php.php`

Comment (obenland) [reply tests/php/core-api/wpcom-endpoints/test_class-wpcom-rest-api-v2-endpoint-external-media.php.php]: Nice catch! Fixed in 3b4a5f5.

**Pull request merged**